kahuna/public/js/components/gr-collections-panel/gr-collections-panel.js

```diff
@@ -37,13 +39,55 @@ grCollectionsPanel.controller('GrCollectionsPanelCtrl', [
     ctrl.selectedCollections = selectedCollections;
 }]);
 
+grCollectionsPanel.factory('collectionsTreeState', ['$window', function($window) {
+    // TODO: Add garbage collection to state.
+    const localStorageKey = 'collectionsTreeOpen';
+    const jsonStr = $window.localStorage.getItem(localStorageKey) || '[]';
+
+    // A little bit of superstition in case this was set weirdly before.
+    let jsonArr = [];
+    try {
+        let jsonArr = JSON.parse(jsonStr);
+        if (!Array.isArray(jsonArr)) {
+            throw new Error(`Invalid ${localStorageKey} json: ${jsonArray}`);
+        }
```

Comment:
I'm not sure if it's a bug or feature, but ES6 (at least the transpilers) scope `const`s in the `try catch` blocks respectively.

So:
```
try {
  const jsonArr = JSON.parse(jsonStr);
} catch(e) {
  const jsonArr = []
}
```
transpiles too:
```JS
try {
    var jsonArr = JSON.parse(jsonStr);
  } catch (e) {
    var jsonArr$__1 = [];
  }
```

> the parent jsonArr

I'm not sure which parent `val` you're talking about?